### PR TITLE
fix: Leverage `get_scorer` and the fact it is a `_BaseScorer`

### LINKED ISSUE #1714: fix(EstimatorReport): sklearn metrics are not labelled correctly in resulting DataFrame
### Describe the bug

See reproducer

### Steps/Code to Reproduce

```python
from sklearn.datasets import load_breast_cancer
from sklearn.linear_model import LogisticRegression
from skore import train_test_split
from skore import EstimatorReport
X, y = load_breast_cancer(return_X_y=True)
split_data = train_test_split(X=X, y=y, random_state=0, as_dict=True)
classifier = LogisticRegression(max_iter=10_000)
report = EstimatorReport(classifier, **split_data)
report.metrics.report_metrics(
     scoring=["neg_log_loss"],
     indicator_favorability=True
)
```

### Expected Behavior

The index of the resulting DataFrame should read "Negative Log Loss", or perhaps "neg_log_loss".

### Actual Behavior

```python
          LogisticRegression Favorability
Metric
Log Loss            0.106768         (↘︎)
```

### Environment

```shell
skore 9bbfdb3
```

### PR DESCRIPTION
**Pull request by glemaitre**
closes https://github.com/probabl-ai/skore/issues/1715
closes #1714
closes https://github.com/probabl-ai/skore/pull/1716
closes https://github.com/probabl-ai/skore/issues/1686

This PR:

- fixes the way we handle scikit-learn scorer names
- update the API documentation to provide the difference between built-in metrics and scikit-learn scorer names
- fixes the tests that are available
- add more tests for metrics that takes additional parameters

### CODE REVIEW

File: skore/src/skore/sklearn/_estimator/metrics_accessor.py [hunk 1]
```diff
@@ -138,14 +143,13 @@ def report_metrics(
...
-        scoring=["neg_log_loss"],
-        indicator_favorability=True)
-                    LogisticRegression Favorability
-        Metric
-        Negative Log Loss      -0.10...        (↘︎)
+        ...     scoring=["f1"], pos_label=1, indicator_favorability=True
+        ... )
+                                  LogisticRegression Favorability
+        Metric   Label / Average
+        F1 Score 1                              0.96...      (↗︎)
```
Comment (auguste-probabl) [inline skore/src/skore/sklearn/_estimator/metrics_accessor.py:152]: ``suggestion
                                  LogisticRegression Favorability
        Metric   Label / Average
        F1 Score               1             0.96...          (↗︎)
```

**Review by auguste-probabl — APPROVED** [review]
Nice work

**Comment by glemaitre** [comment]
The rendering is fine.
